src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -624,24 +639,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
     End Class
 
-    Friend Partial Class BoundUserDefinedUnaryOperator
+    Partial Friend Class BoundUserDefinedUnaryOperator
         Implements IUnaryOperatorExpression
 
         Private ReadOnly Property IHasOperatorMethodExpression_OperatorMethod As IMethodSymbol Implements IHasOperatorMethodExpression.OperatorMethod
             Get
-                Return Me.Call.Method
+                Return If(Me.UnderlyingExpression.Kind = BoundKind.Call, Me.Call.Method, Nothing)
             End Get
         End Property
 
         Private ReadOnly Property IHasOperatorMethodExpression_UsesOperatorMethod As Boolean Implements IHasOperatorMethodExpression.UsesOperatorMethod
             Get
-                Return True
+                Return IHasOperatorMethodExpression_OperatorMethod IsNot Nothing
             End Get
         End Property
 
         Private ReadOnly Property IUnaryOperatorExpression_Operand As IOperation Implements IUnaryOperatorExpression.Operand
             Get
-                Return Me.Operand
+                If Me.UnderlyingExpression.Kind = BoundKind.Call Then
+                    Return Me.Operand
+                Else
+                    Return GetChildOfBadExpression(Me.UnderlyingExpression, 0)
```

Comment:
Try to add throw in the relevant part of implementation and run semantics tests.